### PR TITLE
Add nix flake

### PR DESCRIPTION
**Pull request by beezow**
Adds:

- Nix builder for humility that is capable of running the full humility test suite.
- Flake that setups the development environment and exposes the humility package.
- CI pipeline that runs on both linux and macos.

You can build humility by running `nix build "git+ssh://git@github.com/rivosinc/humility.git?ref=dev/drew/nix_ci"`.  The output will be in the `result` folder.  You can also enter an environment with humility available by running `nix shell "git+ssh://git@github.com/rivosinc/humility.git?ref=dev/drew/nix_ci"`

### CODE REVIEW

**Comment by beezow** [comment]
Can someone verify this works for them on a their mac?  I have it running on the github CI, but figure a sanity check is good.

**Review by FawazTirmizi — DISMISSED** [review]

**Comment by ted-logan** [comment]
> Can someone verify this works for them on a their mac?

Looks good to me on my mac, to the extent that "printing nothing without an error" seems to be success.

**Comment by beezow** [comment]
> > Can someone verify this works for them on a their mac?
> 
> Looks good to me on my mac, to the extent that "printing nothing without an error" seems to be success.

Yep, that generally means success. You can try `nix shell "git+ssh://git@github.com/rivosinc/humility.git?ref=dev/drew/nix_ci"` followed by something like `humility manifest` to ensure it runs.

File: humility.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      cargo
+    ]
+    ++ lib.optionals stdenv.isDarwin [AppKit];
+
+  buildInputs =
+    [
+      libusb1
+      cargo-readme
+    ]
+    ++ lib.optionals stdenv.isLinux [systemd];
```
Comment (picostove) [inline humility.nix:35]: what's this needed for? udev?

Comment (beezow) [reply humility.nix:35]: Yea, I added a comment to reflect this.

**Review by picostove — APPROVED** [review]

**Pull request merged**